spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkFilters.java

```diff
@@ -161,10 +161,13 @@ public static Expression convert(Filter filter) {
 
         case IN:
           In inFilter = (In) filter;
+          Preconditions.checkArgument(
+              Stream.of(inFilter.values()).noneMatch(Objects::isNull),
+              "Expression can not be converted (in is not null-safe): %s",
+              filter);
```

Comment:
This should return null, not throw an exception. We don't ever want to fail conversion, just return `null` so the caller doesn't push an expression down.

Reply:
Updated